x-pack/plugin/profiler/src/main/java/org/elasticsearch/xpack/profiler/GetProfilingRequest.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+package org.elasticsearch.xpack.profiler;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.IndicesRequest;
+import org.elasticsearch.action.support.IndicesOptions;
+import org.elasticsearch.common.ParsingException;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.tasks.CancellableTask;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.tasks.TaskId;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+import static org.elasticsearch.action.ValidateActions.addValidationError;
+import static org.elasticsearch.index.query.AbstractQueryBuilder.parseTopLevelQuery;
+
+/**
+ * A request to get profiling details
+ */
+public class GetProfilingRequest extends ActionRequest implements IndicesRequest {
+    public static final ParseField QUERY_FIELD = new ParseField("query");
+    public static final ParseField SAMPLE_SIZE_FIELD = new ParseField("sample_size");
+
+    private QueryBuilder query;
+
+    private Integer sampleSize;
+
+    public GetProfilingRequest() {
+        this(null, null);
+    }
+
+    public GetProfilingRequest(Integer sampleSize, QueryBuilder query) {
+        this.sampleSize = sampleSize;
+        this.query = query;
+    }
+
+    public GetProfilingRequest(StreamInput in) throws IOException {
+        this.query = in.readOptionalNamedWriteable(QueryBuilder.class);
+        this.sampleSize = in.readOptionalInt();
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeOptionalNamedWriteable(query);
+        out.writeOptionalInt(sampleSize);
+    }
+
+    public Integer getSampleSize() {
+        return sampleSize;
+    }
+
+    public QueryBuilder getQuery() {
+        return query;
+    }
+
+    public void parseXContent(XContentParser parser) throws IOException {
+        if (parser == null) {
```

Comment:
Yes, indeed. This is because I've used `RestRequest#withContentOrSourceParamParserOrNull()` before. I've now removed this null check and changed to `RestRequest#applyContentParser()` in https://github.com/elastic/elasticsearch/pull/91640/commits/62b9fc891316972d7186c7af8978a7be0fe0ef39 which does not require this check.